src/engine/math_util.h

```diff
@@ -449,7 +449,57 @@ extern f32 gSineTable[];
     }                                   \
 }
 
-#define ABS(x)  (((x) > 0) ? (x) : -(x))
+// Float lerp.
+// Return the value between [from, to] based on f's value between [0.0, 1.0].
+ALWAYS_INLINE f32 lerpf(f32 from, f32 to, f32 f) {
+    return (from + (f * (to - from)));
+}
+
+// Precise float lerp.
+// Return the value between [from, to] based on f's value between [0.0, 1.0].
+// Sloer but more precise than the regular float lerp.
+ALWAYS_INLINE f32 lerpf_precise(f32 from, f32 to, f32 f) {
+    return ((from * (1.0f - f)) + (to * f));
+}
+
+// Integer lerp.
+// Return the value between [from, to] based on f's value between [0, 256].
+// Faster than float lerps.
+ALWAYS_INLINE s32 lerpi(s32 from, s32 to, u32 f) {
+    return (((f * (to - from)) >> 8) + from);
+}
+
+// Helper function for smoothstop.
+ALWAYS_INLINE f32 to_smoothstop(f32 x) {
+    f32 sq = sqr(1.0f - x);
+    return (1.0f - sq);
+}
+
+// Commonly known as ease-in.
+ALWAYS_INLINE f32 smoothstart(f32 from, f32 to, f32 amount) {
+    return lerpf(from, to, sqr(amount));
+}
+
+// Commonly known as ease-out.
+ALWAYS_INLINE f32 smoothstop(f32 from, f32 to, f32 amount) {
+    return lerpf(from, to, to_smoothstop(amount));
+}
+
+// Commonly known as ease-in-out.
+ALWAYS_INLINE f32 smoothstep(f32 from, f32 to, f32 amount) {
+    amount = sqr(amount) * (3.0f - (2.0f * amount));
+
+    return lerpf(from, to, amount);
+}
+
+// Remaps a number from one range to another.
+// Return the value between [fromB, toB] based on X's value between [fromA, toA].
+// Equivalent to lerp but with a custom range for f.
+ALWAYS_INLINE f32 remap(f32 f, f32 fromA, f32 toA, f32 fromB, f32 toB) {
+    return ((((f - fromA) / (toA - fromA)) * (toB - fromB)) + fromB);
+}
+
+#define ABS(x) (((x) > 0) ? (x) : -(x))
```

Comment:
a lot of these are going to conflict with the math utils PR and i am not a fan of them. is a single one of these actually being used or are you just trying to add even more macros for the sake of it

Reply:
actually i didnt notice that they're just being moved from below, but regardless i think these are already gone and idk why you bothered to just move them

Reply:
This would have been done before the math util cleanup PR. These are moved back now.